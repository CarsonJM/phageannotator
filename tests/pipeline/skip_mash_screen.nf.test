nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "test_skip_mash_screen"
    tag "skip_mash_screen"
    tag "pipeline"

    test("Skip mash screen") {

        when {
            params {
                outdir              = "$outputDir"
                skip_mash_screen    = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions") },
                { assert new File("$outputDir/fastqc/test1_1_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/test1_2_fastqc.html").exists() },
                { assert snapshot(path("$outputDir/multiqc/multiqc_data/multiqc_fastqc.txt"),
                                path("$outputDir/multiqc/multiqc_data/multiqc_general_stats.txt")).match("multiqc") }
            )
        }

    }

}
