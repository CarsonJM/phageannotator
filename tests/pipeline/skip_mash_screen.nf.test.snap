{
    "software_versions": {
        "content": [
            "{CUSTOM_DUMPSOFTWAREVERSIONS={python=3.11.0, yaml=6.0}, FASTQC={fastqc=0.11.9}, Workflow={nf-core/phageannotator=1.0dev}}"
        ],
        "timestamp": "2023-07-27T19:11:33+0000"
    },
    "multiqc": {
        "content": [
            "multiqc_fastqc.txt:md5,953352f4a0f1bac9058125b8f04e9926",
            "multiqc_general_stats.txt:md5,671fea3ac1c370f05baed7350eac1477"
        ],
        "timestamp": "2023-07-27T19:11:33+0000"
    }
}